documentation/monitoring.md

```diff
@@ -0,0 +1,49 @@
+Monitoring
+------------
+
+New Relic is a SaaS tool for monitoring and alerting. Ocim supports using `New Relic Synthetics` and `New Relic Alerts` for monitoring the instances.
+
+On Alerts, policies can be created to control the alerting behaviour. Alerts has two main components:
+* **Alert Policies**: An alert policy is a group of one or more alert conditions. A policy has two settings that apply to all of its conditions: incident preference and notification channels. A policy should be created before adding conditions to it.
+* **Alert Conditions**: its a combination of monitored data source, such as a Synthetics Monitor, and thresholds that define the behavior that will be considered a violation. Synthetics only reports if the pings succeeded or not, so our threshold is to check if the ping failed.
+
+Each **Alert Policy** contains one or more **Alert Conditions** that can be configured to modify the alerting behaviour.
+
+For example, on Open edX instances, we create a alert policy with 1 alert per policy, and add an alert condition for each service (LMS, Studio, Preview).
+That way, if any service is down, the monitor will trigger an alert. But if there's more than one service failing pings, only a single alert will be fired, to prevent firing a massive amount of alert emails when an entire instance goes down or there are OVH/AWS network issues.
+
+The image below shows the concept behind Alerts.
+
+![New Relic Alerts](./images/newrelic-alerts.png)
+
+A more in deph explanation of how New Relic Alerts work can be found [here](https://docs.newrelic.com/docs/alerts/new-relic-alerts/getting-started/new-relic-alerts-concepts-workflow).
+
+## How it works on Ocim
+
+Ocim automatically sets up monitoring for each instance, and creates all the necessary NewRelic resources for alerting. It does that by running `.enable_monitoring()` on an instance during provisioning.
+
+Additionally, the monitoring set up methods can be called manually to disable monitoring or rebuild NewRelic assets changed due to changes in configuration.
+
+* `.enable_monitoring()`: Set ups resources on New Relic to enable monitoring, if the resources already exist, overwrite them with new settings.
+* `.disable_monitoring()`: Deprovision all resources on New Relics related to that instance, including Alert Policies, Alert Conditions, Notification channels (if not shared) and Synthetics monitors. This is used when an instance is archived and might be useful for debugging.
+
+## Deploying production instances
+
+For production instances, some manual setup is required to alert to other notification channels (such as PagerDuty or a email list).
+
+1. Using Django Shell (`make shell`), find the instance you want to add monitoring to.
```

Comment:
@giovannicimolin, this can be done using the Django admin interface.

Reply:
@giovannicimolin, it needs a redeployment. We always do a redeployment to apply any changes made to an instance's configuration. This might be a lot of work than running `enable_monitoring()`, but it is much safer because there always logs that capture what was done.

Reply:
:+1: